### ISSUE TITLE
adding init_container  to a deployment causes unnecessary replacement

### ISSUE THREAD

**Issue by redbaron** [hashicorp/terraform-provider-kubernetes#951]
### Terraform Version and Provider Version
```
Terraform v0.12.28
+ provider.aws v2.70.0
+ provider.google v3.32.0
+ provider.google-beta v3.30.0
+ provider.http v1.2.0
+ provider.kubernetes v1.12.0
+ provider.null v2.1.2
+ provider.random v2.3.0
```

### Affected Resource(s)
`kubernetes_deployment`

### Terraform Configuration Files
```
resource "kubernetes_deployment" "flux" {
...
        init_container {
          name  = "tools"
          image = "alpine:3.12"
          command = [
            "/bin/sh", "-c", "wget -o /tools/kubectl https://storage.googleapis.com/kubernetes-release/release/v1.19.0-rc.4/bin/linux/amd64/kubectl && chmod +x /tools/kubectl"
          ]

          volume_mount {
            name       = "tools"
            mount_path = "/tools"
          }

        }
}
```

### Expected Behavior
Resource is changed

### Actual Behavior

Resource is replaced

### Steps to Reproduce

1. Create `kubernetes_deployment` without `init_container`
2. add `init_container` section and check plan 

### Community Note
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jrhouston** [comment]
This seems to be happening because we are re-using the schema for container in both deployment and pod, and for pod changing a container is a ForceNew operation - but for Deployment it is not as the pod is just a template.

This is related to #955, see my comment here: https://github.com/hashicorp/terraform-provider-kubernetes/issues/955#issuecomment-681022027

**Comment by shadowfax-chc** [comment]
This is also impacting other resources that use templates, such as cronjobs.

**Comment by dak1n1** [comment]
Thanks for this bug report! We released a fix for the init_container issue in 2.0. We're tracking the cronjob issue in https://github.com/hashicorp/terraform-provider-kubernetes/issues/1139, so I'll close this ticket in favor of #1139.
